server/controllers/verificationController.js

```diff
@@ -0,0 +1,91 @@
+    let retailerId = params.retailerId;
+    let randomPIN = Math.floor(100000 + Math.random() * 900000);
+    let toNumber = body.phoneNumber;
+    let toEmail = body.email;
+
+    Retailer.findByIdAndUpdate(
+      { _id: retailerId },
+      { verificationPIN: randomPIN },
+      { new: true }
+    )
+      .select("verificationPIN retailerName")
+      .then((results) => {
+        console.log("verificationController: ", results);
+        let retailerName = results.retailerName;
+        let verificationPIN = results.verificationPIN;
+        if (verificationType === "text") {
+          client.messages
+            .create({
+              body: `Hello ${retailerName}, your verification PIN is ${verificationPIN}`,
+              from: twilioNumber,
+              to: toNumber,
+            })
+            .then((message) => console.log(message))
+            .catch((err) => console.log(err));
+        } else if (verificationType === "call") {
+          client.calls
+            .create({
+              to: toNumber,
+              from: twilioNumber,
+              twiml: `<Response><Say>Hello ${retailerName}, this is a call from Queue App. Your verification pin is ${verificationPIN
+                .toString()
+                .split("")
+                .join(
+                  ", "
+                )}. Please verify your pin in our app. Again, your verification pin is ${verificationPIN
+                .toString()
+                .split("")
+                .join(", ")}. Thank you.</Say></Response>`,
+            })
+            .then((message) => console.log(message))
+            .catch((err) => console.log(err));
+        } else if (verificationType === "email") {
+          sgMail
+            .send({
+              to: `${toEmail}`,
+              from: "admin@thequeue.dev",
+              subject: "Queue App Verification",
+              text: `Hello ${retailerName}, your PIN is ${verificationPIN}. Please verify through our app. Thank you!`,
+              html: `<body><h1>Hello ${retailerName},</h1><p>your PIN is ${verificationPIN}. Please verify through our app. Thank you!</p></body>`,
+            })
+            .then((message) => console.log(message))
+            .catch((err) => console.log(err));
+        }
+      });
+  },
+  checkPIN({ params, body }, res) {
+    let retailerId = params.retailerId;
+    let enteredPIN = body.enteredPIN;
+    Retailer.findById({ _id: retailerId }).then((results) => {
+      let verificationPIN = results.verificationPIN;
+      if (enteredPIN === verificationPIN) {
+        return Retailer.findByIdAndUpdate(
+          { _id: retailerId },
+          { verificationStatus: true },
+          { new: true }
+        ).then((results) => {
+          res
+            .status(200)
+            .json({ message: "Your business has been successfully verified." });
+        });
+      }
+      res.status(200).json({ error: "Sorry the PIN does not match." });
+    });
```

Comment:
what if here we did a

```js
Retailer.findOne({
_id: retailerID,
verificationPIN: enteredPIN
}).then(results =>{
if(!results){
return res.status(200).json({ error: "Sorry the PIN does not match." });
}

return res.status(200).json({message: "Your business has been successfully verified."});

}).catch(err=>{
console.log(err);
return res.status(500).json({error: err});
```

Reply:
the reason why I did two calls to the database was because I first had to grab the PIN from the retailer model, compare it, then update the verificationStatus to true.